### DISCUSSION TITLE
Switch from unmaintained lash library to the more active ladish one

### DISCUSSION THREAD

**Discussion by GvMariani** [FluidSynth/fluidsynth#1122] [Ideas]
The lash library, used for session management with jack sound server, is very old (last release in 2009) and unmaintained (no commit in the upstream git since July 2009).
Luckily there is a viable replacement: the Nedko Arnaudov's LADI Session Handler (https://github.com/LADI/ladish). It's a complete rewrite of lash and is much more active (last git commit was in February 2021); it's already used by Carla and Cadence packages from Kxstudio project. It also sports an handy compatibility layer (liblash): all it should be needed to use it is changing the references to lash and its headers in those for the ladish compatibility layer and its headers.

The test patch attached simply does this, without bothering to change "lash" in "ladish" in the messages: it builds and runs OK for simple use on my system...
[fluidsynth-2.2.7-use-ladish-instead-of-lash.txt](https://github.com/FluidSynth/fluidsynth/files/8968660/fluidsynth-2.2.7-use-ladish-instead-of-lash.txt)

Can the switch be considered?

**Comment by derselbst** [comment]
> Can the switch be considered?

Sure. I never used lash myself though. I thought about removing it. But since you seem to find it useful and there's an adequate replacement, we could probably switch to ladish in fluidsynth 2.3.0. The patch looks super simple. You're welcome to rebase it against master and draft a PR.

**Reply by GvMariani** [reply]
> The patch looks super simple.
Yep: it lacks any lash -> ladish change in the variable names and user messages, but that should be a simple search-and-replace work...
I will try to produce a more complete PR.
